### DISCUSSION TITLE
What is the difference between add a subjet and add a subject example

### DISCUSSION THREAD

**Discussion by leccyril** [exadel-inc/CompreFace#813] [General] [answered]
Hello, 

We tried to define exactly what is the difference between method add subject and method add subject example described here : 

[add subject example](https://documenter.getpostman.com/view/17578263/UUxzAnde#9ba2be80-74ca-44e5-aed1-4941ad8ac600)

and

[ add subject ](https://documenter.getpostman.com/view/17578263/UUxzAnde#e560171c-ad29-40a7-a4d5-3afcea1d8b79)

As we can see in both we can add several pictures for 1 subject, so what is the subtility between 2 methods (maybe a use case can be helpful)

Thanks !

**Comment by pospielov** [comment, marked as answer]
The subject example is a face example. This is one example of the face.
The subject is a collection of subject examples. 
You can see it as a subject is a table in DB and subject examples are the rows. (We do not create a table for each subject thought)

When you create a subject, it's empty. 
For facial recognition, we use only subject examples. So empty subject doesn't take part in facial recognition.

When you create a subject example, we check if there is a subject, then we just add an example to the subject. If there is no such subject, we create one. 

Basically, we added work with the subject to simplify work on our UI. But as you are going to have your own UI, probably those endpoints will help you too.

**Reply by leccyril** [reply]
sorry i did not see that in subject create method there is no picture to upload.

great !
